### PR TITLE
Mute function added

### PR DESCRIPTION
**Pull request by AardCrowley**
Feel free to clean up any code as necessary, or leave as-is.

### CODE REVIEW

**Comment by AardCrowley** [comment]
Had to add a conditional to chat echo due to the replay buffer.

**Comment by AardCrowley** [comment]
Moved a conditional for isMuted. It should have occurred before the edit mode check, not on the same line.

File: MUSHclient/worlds/plugins/aard_channels_fiendish.xml [hunk 1]
```diff
@@ -29,6 +31,33 @@ USAGE:
...
+
+<trigger
+   enabled="y"
+   name="ignoreLimit"
+   match="^You are already ignoring 10 people\.$"
+   regexp="y"
+   script="metIgnoreLimit"
+   omit_from_output="y"
+   sequence="100"
+></trigger>
```
Comment (fiendish) [inline MUSHclient/worlds/plugins/aard_channels_fiendish.xml]: Why?

File: MUSHclient/worlds/plugins/aard_channels_fiendish.xml [hunk 2]
```diff
@@ -134,6 +180,124 @@ require "scrollbar"
...
+else
+   muteList = {}
+   SetVariable("muteList", serialize.save("muteList"))
+end
+
+function muteName(name, line, wildcards)
+
+   local playerName = wildcards[1]:lower():gsub("^%l", string.upper)
+
+	if playerName:upper() == "CLEAR" then
```
Comment (fiendish) [inline MUSHclient/worlds/plugins/aard_channels_fiendish.xml]: The indentation is chaotic. Can you clean it up to 3 spaces?

File: MUSHclient/worlds/plugins/aard_channels_fiendish.xml [hunk 3]
```diff
@@ -134,6 +180,124 @@ require "scrollbar"
...
+
+
+function startIgnoreList()
+	EnableTrigger("ignoreLimit", false)
+	EnableTrigger("ignoreEnd")
+end
+
+function endIgnoreList()
+	Note("")
+	EnableTrigger("ignoreStart", false)
```
Comment (fiendish) [inline MUSHclient/worlds/plugins/aard_channels_fiendish.xml]: btw, ignoreStart can disable itself.

File: MUSHclient/worlds/plugins/aard_channels_fiendish.xml [hunk 4]
```diff
@@ -134,6 +180,124 @@ require "scrollbar"
...
+		   break
+	   end
+   end
+	return playerInList
+end
+
+function encounterLimit()
+   EnableTrigger("ignoreLimit", false)
+   metIgnoreLimit(playerCapture)
+end
```
Comment (fiendish) [inline MUSHclient/worlds/plugins/aard_channels_fiendish.xml]: This doesn't appear to be used.

**Review by fiendish — CHANGES_REQUESTED** [review]
the aliases and storage need to either be in the echo plugin or in some other plugin because it's possible to have multiple window plugins active simultaneouly

File: MUSHclient/worlds/plugins/aard_channels_fiendish.xml (hunk 1 above)
Comment (AardCrowley) [reply MUSHclient/worlds/plugins/aard_channels_fiendish.xml]: If someone tries to ignore a person, it seems obvious to me that they do not want to hear from them, so it'll automatically attempt to silence the person instead.

File: MUSHclient/worlds/plugins/aard_channels_fiendish.xml (hunk 3 above)
Comment (AardCrowley) [reply MUSHclient/worlds/plugins/aard_channels_fiendish.xml]: Moved the disable to ignoreStart.

File: MUSHclient/worlds/plugins/aard_channels_fiendish.xml (hunk 4 above)
Comment (AardCrowley) [reply MUSHclient/worlds/plugins/aard_channels_fiendish.xml]: Removed this function. I forget why I even bothered with it, but I think it was a bit more involved before in testing phases.

File: MUSHclient/worlds/plugins/aard_channels_fiendish.xml (hunk 2 above)
Comment (AardCrowley) [reply MUSHclient/worlds/plugins/aard_channels_fiendish.xml]: Cleaned up the indentation.

File: MUSHclient/worlds/plugins/aard_chat_echo.xml [hunk 5]
```diff
@@ -239,6 +408,8 @@ function OnPluginBroadcast (msg, id, name, text)
          local gmcp_comm = gmcp("comm.channel")
          local msg = gmcp_comm.msg or ""
          local chan = gmcp_comm.chan or ""
+         local player = gmcp_comm.player or ""
+         local res, isMuted = CallPlugin("b555825a4a5700c35fa80780", "checkIfMuted", player)
```
Comment (fiendish) [inline MUSHclient/worlds/plugins/aard_chat_echo.xml]: checkIfMuted does not exist in b555825a4a5700c35fa80780

Comment (AardCrowley) [reply MUSHclient/worlds/plugins/aard_chat_echo.xml]: Such a silly mistake to miss.

**Comment by AardCrowley** [comment]
Is there anything else that I need to take care of before this is pulled over? I think I've addressed all the other issues.

**Comment by AardCrowley** [comment]
Updated Soundpack to include Epics channel per Tymme's request. It currently uses the 'death.wav' sound, but if someone wants to find an 'epics' wav, they can add that to the sounds directory.

**Comment by fiendish** [comment]
bless. sorry it's taking so long.

**Comment by AardCrowley** [comment]
> bless. sorry it's taking so long.

No worries! Busy lives, real life, and bug testing all play a part. I know you'll get around to it when you can get around to it. It's no rush - I have it on my end, and I'm not missing out on any updates otherwise.

**Comment by fiendish** [comment]
Please try https://github.com/fiendish/aardwolfclientpackage/tree/development
I rewrote the code, so just going to close the PR.

**Pull request closed without merging**